util.c

```diff
@@ -346,3 +347,32 @@ init_random(void)
 	if (rf != -1)
 		close(rf);
 }
+
+void cleanUp(void)
+{
+	deltmp();
+	free_all_auth_entries();
```

Comment:
not needed.  we don't free on exit.  You can just keep atexit(deltmp)

Reply:
ok